### PR TITLE
update propolis and crucible deps

### PR DESCRIPTION
**Pull request by ahl**
lockstep told me there was nothing else to be done

### CODE REVIEW

File: package-manifest.toml [hunk 1]
```diff
@@ -84,7 +84,7 @@ zone = true
 # 3. Use type = "manual" instead of the "prebuilt"
 type = "prebuilt"
 repo = "crucible"
-commit = "d412b2be7ef8c0658bf5603f7fd746f6972e4666"
+commit = "8ea317a9dfee65d7c018a1fda89fdec63a560ef3"
 # The SHA256 digest is automatically posted to:
 # https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/<commit>/crucible.sha256.txt
 sha256 = "f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb"
```
Comment (jmpesp) [inline package-manifest.toml:90]: it's surprising to me that this didn't change, given the commit rev did...?

Comment (jmpesp) [reply package-manifest.toml:90]:
```
$ curl -sq https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/d412b2be7ef8c0658bf5603f7fd746f6972e4666/crucible.sha256.txt
f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb

$ curl -sq https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/8ea317a9dfee65d7c018a1fda89fdec63a560ef3/crucible.sha256.txt
f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb
```

seems legit

Comment (bnaecker) [reply package-manifest.toml:90]: It looks like the actual Buildomat output file, not the published one, is different. See [this URL](https://buildomat.eng.oxide.computer/wg/0/artefact/01G36RYYRCV9GJYYY6ASP8Q7BM/2YCR5nTIIhxLamd83XyWBDLuNiXJaH6JgaZmQDHQ9H2TT03G/01G36RZ8AY573DDP0MGZVMZJAV/01G36SWBYV892FGCFKJPXMTF79/crucible.sha256.txt). Maybe the publishing step grabbed the wrong file somehow?

Comment (bnaecker) [reply package-manifest.toml:90]: I'm a bit confused. The Buildomat output at the bottom of [this job output](https://buildomat.eng.oxide.computer/wg/0/details/01G36RYYRCV9GJYYY6ASP8Q7BM/2YCR5nTIIhxLamd83XyWBDLuNiXJaH6JgaZmQDHQ9H2TT03G/01G36RZ8AY573DDP0MGZVMZJAV) seems to show the digest being computed with:

```
digest -a sha256 crucible.tar.gz
```

As far as I know, that echoes the SHA, but doesn't redirect anything a file. Shouldn't it do so? Where does the file at `crucible.sha256.txt` that's being uploaded in the following step come from?

Comment (ahl) [reply package-manifest.toml:90]: haha... well ... mystery solved?

Comment (jmpesp) [reply package-manifest.toml:90]: I see the same thing:

```
$ curl -sq https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/d412b2be7ef8c0658bf5603f7fd746f6972e4666/crucible.tar.gz | sha256sum
f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb  -

$ curl -sq https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/8ea317a9dfee65d7c018a1fda89fdec63a560ef3/crucible.tar.gz | sha256sum
f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb  -
```

Comment (ahl) [reply package-manifest.toml:90]: That file comes from here:
https://github.com/oxidecomputer/crucible/blob/main/.github/buildomat/jobs/image.sh#L43

Comment (jmpesp) [reply package-manifest.toml:90]: In fact, looking at the last 20 commits of crucible, it seems like the image's sha256 does not change every time, which is surprising:

```
$ for commit in $(git log | grep commit | head -n20 | cut -d' ' -f2); do echo "${commit} $(curl -sq https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/${commit}/crucible.tar.gz | sha256sum)"; done;
8ea317a9dfee65d7c018a1fda89fdec63a560ef3 f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb  -
5b6a40978f8f5bf57eb5781f833cf6b2d2b99359 f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb  -
d412b2be7ef8c0658bf5603f7fd746f6972e4666 f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb  -
c708b8024060a06791485fb3712ad839b64351f4 f85bb22797bab58d128c8dd168a2e6ba57feee284bdc9d0adf282197cb45cccb  -
1c89c283ddad0825ebf1b45a2b049c5dc86a9346 3c593037f51df8c2ce57bcec7824eb278f853c44f5ff778757b64274f25b68c2  -
bad407b43de03395be2bbe2f81bd4a895f5a5e97 3c593037f51df8c2ce57bcec7824eb278f853c44f5ff778757b64274f25b68c2  -
6956ef611a2f32051ef9ab620154ae1eb29de880 3c593037f51df8c2ce57bcec7824eb278f853c44f5ff778757b64274f25b68c2  -
212183cf577befc40b20d74c4e7895e0fee3cb5a d205188f8ac5899d9ad7317d1ffd54172b96e68b239a0a6b91775de22b57f94b  -
3cd5ddef060d09f14e1f4a862227dda9c93c820b b3335b4b3252c2158817825f6bde311f8b50d7cc7b87932316aeb29f76d36ae1  -
75ff03de32096fa71d004d306a513feebeba7499 1b1264efb6864c45dd7f8212ee30000533920eb09085f6e5625733fbfaa8baba  -
5a9faeeaffc318c7f8e9354b81aab98728124426 d94368130cc2ea6607c313a99c54c6aa050f46a9c24832227f3009286cf183e9  -
cd9e82123a0261ef3efb148f3a10e77737fc4bb5 360fea6d7d5e482c51a35d676484f48951a83c6cd08f1376c1160e864f36d8f0  -
38236f56a121049de4709e1d7dcde88bba8097c8 f1195a39ac0ef3935bec9ca6e329f3c4a0b934896fa67138dbf5b608ee29440f  -
023f3711748d99f0b478ac08887edd8382b477df 037caa7774af8ea36731af373f16a0b228f3e2b258f5a393b1ddc4764a3fc1c8  -
f1002c3d9057d339bbc5c4fd8eedf6b57f97c318 d336e7b3fc431a6d3ed9f7307c653bbd646bebf9839f6a6c089d74af9673dd87  -
cb6f71b6df8045a074bb1830c14b252b0193d191 d336e7b3fc431a6d3ed9f7307c653bbd646bebf9839f6a6c089d74af9673dd87  -
e4df5eac7b5f5e7110f285ff279c8353473b6d48 d336e7b3fc431a6d3ed9f7307c653bbd646bebf9839f6a6c089d74af9673dd87  -
695762041979ac674fdac747068b25282b615398 d336e7b3fc431a6d3ed9f7307c653bbd646bebf9839f6a6c089d74af9673dd87  -
1be2f4b095160f71611932f7846fd8fea535dc60 610f4dbe6f38c4f3bf2394a07ab05d5680d95d1528a19b796d21a91f748db06a  -
dc6f870f893a3da1b47877143e2bc9e3aef8113a 610f4dbe6f38c4f3bf2394a07ab05d5680d95d1528a19b796d21a91f748db06a  -
```

**Review by bnaecker — APPROVED** [review]

File: package-manifest.toml (hunk 1 above)
Comment (smklein) [reply package-manifest.toml:90]: The SHA256 references the generated zone image, not the entire source tree at a particular commit.

If changes were made to the repo that did not modify the downstairs zone image, I would expect it to stay the same.

Comment (bnaecker) [reply package-manifest.toml:90]: That seems right. This last commit that @ahl added didn't change anything that's packaged into the zone tarball, but some of [those previous commits](https://github.com/oxidecomputer/crucible/commit/c708b8024060a06791485fb3712ad839b64351f4) _do_ change things like the agent or downstairs binaries yet result in the SHA. I'd expect that to change the digest of the zone image, but I could be missing something.

Comment (ahl) [reply package-manifest.toml:90]: I didn't realize that build were purely deterministic.

Comment (jclulow) [reply package-manifest.toml:90]: That is indeed quite impressive

**Pull request merged**